src/model/keychain.js

```diff
@@ -136,3 +143,86 @@ export async function hasKey(key) {
   }
   return null;
 }
+
+export async function restoreBackupIntoKeychain(backedUpData) {
+  // Access control config per each type of key
+  const publicAccessControlOptions = {
+    accessible: ACCESSIBLE.ALWAYS_THIS_DEVICE_ONLY,
+  };
+
+  let privateAccessControlOptions = {};
+  const canAuthenticate = await canImplyAuthentication({
+    authenticationType: AUTHENTICATION_TYPE.DEVICE_PASSCODE_OR_BIOMETRICS,
+  });
+
+  let isSimulator = false;
+
+  if (canAuthenticate) {
+    isSimulator = __DEV__ && (await DeviceInfo.isEmulator());
+  }
+  if (canAuthenticate && !isSimulator) {
+    privateAccessControlOptions = {
+      accessControl: ACCESS_CONTROL.USER_PRESENCE,
+      accessible: ACCESSIBLE.WHEN_UNLOCKED_THIS_DEVICE_ONLY,
+    };
+  }
+
+  Object.keys(backedUpData).forEach(async key => {
```

Comment:
Our favorite - forEach and async will not behave exactly as you expect this way, in case you want to be clearer about what you are actually awaiting.

Reply:
Hm and given that you want to save the seedPhraseMigratedKey AFTER you've saved everything else, probably should not do the forEach and async mixing